### PR TITLE
Added controller method and service method to be able to create user at a given location

### PR DESCRIPTION
**Pull request by nomss**
Added controller method and service method to be able to create user at a given location

### CODE REVIEW

File: bl-service/src/main/java/com/ayeeti/blservice/location/LocationMapper.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            return null;
+        }
+        return new LocationDTO(
+                location.getId(),
+                location.getAirportName(),
+                location.getAirportCode()
+        );
+    }
+
+//    public static Location mapLocationRequestToLocation(LocationRequest locationRequest) {
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/location/LocationMapper.java]: Remove unused code

File: bl-service/src/main/java/com/ayeeti/blservice/location/requests/LocationRequest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Builder
+@Getter
+@Setter
+@AllArgsConstructor
+public class LocationRequest {
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/location/requests/LocationRequest.java:12]: Not sure why you used class here. We can use record here as for all other Requests.

File: bl-service/src/main/java/com/ayeeti/blservice/user/User.java [hunk 3]
```diff
@@ -32,6 +31,6 @@ public class User {
             joinColumns = @JoinColumn(name = "user_id"),
             inverseJoinColumns = @JoinColumn(name = "location_id")
     )
-    private List<Location> locations = new ArrayList<>();
+    private Set<Location> locations = new HashSet<>();
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/user/User.java:34]: Why did you change the type?

File: bl-service/src/main/java/com/ayeeti/blservice/user/UserController.java [hunk 4]
```diff
@@ -34,4 +35,14 @@ public UserDTO updateUser(@RequestBody UpdateUserRequest userUpdateUserNameReque
         return userService.updateUser(userUpdateUserNameRequest.userId(), userUpdateUserNameRequest.newUserName());
     }
 
+    @PostMapping("/createUserAtLocation")
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/user/UserController.java]: Again you are not naming your endpoints mapping correctly. We've been over this and I explained it 2 times already.

File: bl-service/src/main/java/com/ayeeti/blservice/user/UserDTO.java [hunk 5]
```diff
@@ -1,6 +1,22 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Set;
+
+@Builder
+@AllArgsConstructor
+@Getter
+@Setter
+public class UserDTO {
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/user/UserDTO.java:15]: Why changing the source code constantly???

File: bl-service/src/main/java/com/ayeeti/blservice/user/requests/CreateUserAtLocation.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Set;
+
+@Builder
+@AllArgsConstructor
+@Getter
+@Setter
+public class CreateUserAtLocation {
```
Comment (tarikmujcic) [inline bl-service/src/main/java/com/ayeeti/blservice/user/requests/CreateUserAtLocation.java]: As any request this should have a request in it's name. e.g CreateUserAtLocationRequest and it should be a record! not a class

File: bl-service/src/test/java/com/ayeeti/blservice/user/UserServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package com.ayeeti.blservice.user;
+
+import com.ayeeti.blservice.location.requests.LocationRequest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.Set;
+
+@SpringBootTest
```
Comment (tarikmujcic) [inline bl-service/src/test/java/com/ayeeti/blservice/user/UserServiceTest.java:10]: Again you are writing tests. Why didn't you write a LocationController? Why is the task not completed?

**Review by tarikmujcic — CHANGES_REQUESTED** [review]
Overall you are doing the changes and I am happy with the progress. Just the problem is, every PR is incomplete. You don't finish the task, but you have the tests which are not the part of the task.

Have the tests, no problem - But please finish the task as well. 

Also, make sure to try and keep the consistency in the code. (e.g if every request is name SomethingRequest you cannot name the next one OtherThing - it should be OtherThingRequest.

File: bl-service/src/test/java/com/ayeeti/blservice/user/UserServiceTest.java (hunk 7 above)
Comment (nomss) [reply bl-service/src/test/java/com/ayeeti/blservice/user/UserServiceTest.java:10]: I didn't see the task. I thought I was doing the create user at a given location. But now I have added the Location controller.

File: bl-service/src/main/java/com/ayeeti/blservice/user/requests/CreateUserAtLocation.java (hunk 6 above)
Comment (nomss) [reply bl-service/src/main/java/com/ayeeti/blservice/user/requests/CreateUserAtLocation.java]: I was having hard time with intellisense with the record so I changed to the class.

File: bl-service/src/main/java/com/ayeeti/blservice/user/UserDTO.java (hunk 5 above)
Comment (nomss) [reply bl-service/src/main/java/com/ayeeti/blservice/user/UserDTO.java:15]: Similar to the comment below, I was having hard time with the record. I needed to be able to give errorMessage dynamically.

File: bl-service/src/main/java/com/ayeeti/blservice/user/UserController.java (hunk 4 above)
Comment (nomss) [reply bl-service/src/main/java/com/ayeeti/blservice/user/UserController.java]: We'll have to go over the naming convention again, I have never been a fan of {id}.
